src/test/scala/rx/lang/scala/observables/AsyncOnSubscribeTests.scala

```diff
@@ -0,0 +1,65 @@
+package rx.lang.scala.observables
+
+import org.junit.Test
+import org.junit.Assert._
+import org.scalatest.junit.JUnitSuite
+import rx.lang.scala.observers.TestSubscriber
+import rx.lang.scala.{Notification, Observable}
+
+class AsyncOnSubscribeTests extends JUnitSuite {
+
+  @Test
+  def testStateful(): Unit = {
+    val last = 2000L
+    val sut = Observable.create(AsyncOnSubscribe(() => 0L)((count,demand) =>
+      if(count > last)
+        (Notification.OnCompleted, count)
+      else {
+        val max = math.max(count + demand, last)
+        val next = Observable.from(count to max)
+        (Notification.OnNext(next), max+1)
+      }
+    ))
+    assertEquals((0L to last).toList, sut.toBlocking.toList)
+  }
```

Comment:
This is a good example, could you please also add something like this to [`RxScalaDemo`](https://github.com/ReactiveX/RxScala/blob/0.x/examples/src/test/scala/examples/RxScalaDemo.scala)? Also, adding (A)SyncOnSubscribe requires some information in `RxScalaDemo` to be updated: `createExampleGood` is not good any more, but only "medium good" :wink: It respects unsubscription, but not backpressure, and using (A)SyncOnSubscribe would be better. Could you please review all examples related to `Observable.create`, and add some comments, and examples for (A)SyncOnSubscribe, so that this is all up to date?